packages/client/src/messages.ts

```diff
@@ -1,39 +1,43 @@
-import { Message } from '@botpress/messaging-base'
+import { Message, uuid } from '@botpress/messaging-base'
 import { BaseClient } from './base'
-import { handleNotFound } from './errors'
 
 export class MessageClient extends BaseClient {
   async create(
-    conversationId: string,
-    authorId: string | undefined,
+    conversationId: uuid,
+    authorId: uuid | undefined,
     payload: any,
-    flags?: { incomingId: string }
+    flags?: { incomingId: uuid }
   ): Promise<Message> {
     return this.deserialize(
       (await this.http.post<Message>('/messages', { conversationId, authorId, payload, incomingId: flags?.incomingId }))
         .data
     )
   }
 
-  async get(id: string): Promise<Message | undefined> {
-    return handleNotFound(async () => {
-      this.deserialize((await this.http.get<Message>(`/messages/${id}`)).data)
-    }, undefined)
+  async get(id: uuid): Promise<Message | undefined> {
+    const message = (await this.http.get<Message>(`/messages/${id}`)).data
+    if (message) {
+      return this.deserialize(message)
+    } else {
+      return undefined
+    }
+  }
+
+  async list(conversationId: uuid, limit?: number): Promise<Message[]> {
+    return (await this.http.get<Message[]>(`/messages/conversation/${conversationId}`, { params: { limit } })).data.map(
+      (x) => this.deserialize(x)
+    )
   }
 
-  async list(conversationId: string, limit: number): Promise<Message[]> {
-    return handleNotFound(async () => {
-      return (await this.http.get<Message[]>('/messages', { params: { conversationId, limit } })).data.map((x) =>
-        this.deserialize(x)
-      )
-    }, [])
+  async delete(id: uuid): Promise<boolean> {
+    return (await this.http.delete<boolean>(`/messages/${id}`)).data
   }
 
-  async delete(filters: { id?: string; conversationId?: string }): Promise<number> {
-    return (await this.http.delete<{ count: number }>('/messages', { params: filters })).data.count
+  async deleteByConversation(conversationId: uuid): Promise<number> {
+    return (await this.http.delete<{ count: number }>(`/messages/conversation/${conversationId}`)).data.count
   }
```

Comment:
Notice here I split the message deletion routes into two. So we don't use query params anymore.